### PR TITLE
New SecurityProviderBannerAlert component

### PR DESCRIPTION
**Pull request by digiwand**
## Explanation

Fixes: https://github.com/MetaMask/MetaMask-planning/issues/558
Blocked By: https://github.com/MetaMask/metamask-extension/issues/19768
Blocks: https://github.com/MetaMask/metamask-extension/issues/19257

We will be replacing the OpenSea security provider warning message. The current SecurityProviderBannerMessage component that displays this has specific logic for OpenSea. This PR creates a new UI component that allows for scalability and general use cases. UI differences from the OpenSea provider warning message include the option to include details, an updated footer copy, and a SecurityTick icon.

### New Component 

Screen Shot 2023-06-29 at 4 44 33 PM

## Manual Testing Steps

1. run storybook - `yarn storybook`
2. view new component - http://localhost:6006/?path=/docs/components-app-securityproviderbanneralert--docs

## Pre-merge author checklist

- [x] I've clearly explained:
  - [x] What problem this PR is solving
  - [x] How this problem was solved
  - [x] How reviewers can test my changes
- [x] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [x] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [x] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

File: ui/components/app/security-provider-banner-alert/security-provider-banner-alert.js
Comment (georgewrmarshall) [inline ui/components/app/security-provider-banner-alert/security-provider-banner-alert.js:46]: Could you use the `description` prop inside the `BannerAlert`? I think it should be `bodyMd`? Our typography is responsive so if you see 14px the design might be using the small screen typography. It should be 16px on larger screens. Also I know a lot of the Text are rendering as `h6` this is probably not the best semantic mark up but I think was a decision instead of getting new contributors to update e2e tests. Probably not the best in hindsight. I think it could just be a `p` tag which is what bodySm renders by default

**Review by georgewrmarshall — COMMENTED** [review]
Looking excellent 💯 just some small suggestions. I'm a bit manic this morning but want to give timely reviews so apologies if the comments are a little uncoherent

File: ui/components/app/security-provider-banner-alert/security-provider-banner-alert.stories.js
Comment (digiwand) [reply ui/components/app/security-provider-banner-alert/security-provider-banner-alert.stories.js:30]: ooh nice! Ty!

File: ui/components/app/security-provider-banner-alert/security-provider-banner-alert.js
Comment (digiwand) [reply ui/components/app/security-provider-banner-alert/security-provider-banner-alert.js]: updated [2e6687e](https://github.com/MetaMask/metamask-extension/pull/19825/commits/2e6687ea9a525567575431d4657d3dc30ccd7417)

File: ui/components/app/security-provider-banner-alert/security-provider-banner-alert.js
Comment (digiwand) [reply ui/components/app/security-provider-banner-alert/security-provider-banner-alert.js]: I see now. `description` is inherited from `BannerBase`. Got it 

I was carrying over code and preserving styles from our OpenSea SecurityProviderMessage component. These are great updates 👍  Thank you!

applied in [647eda9](https://github.com/MetaMask/metamask-extension/pull/19825/commits/647eda970ad7c07a4c4a032db766d2aab10cfc73) and [a7fdeee](https://github.com/MetaMask/metamask-extension/pull/19825/commits/a7fdeee4190606a17d9a6d1bf685cdecc7ededc5)

I also adjusted the Disclosure component text size here https://github.com/MetaMask/metamask-extension/pull/19770/commits/8ac92488697a27152e00e9c50d1fe0d23ff95573

One thing using the `description` param is that is a lack of padding between the title and description which does not match the figma design. Any suggestions? @georgewrmarshall

Comment (digiwand) [reply ui/components/app/security-provider-banner-alert/security-provider-banner-alert.js]: before: 
Screen Shot 2023-07-10 at 9 15 35 PM

after using `description` param:
Screen Shot 2023-07-12 at 12 08 47 AM

File: shared/constants/security-provider.ts
Comment (cryptotavares) [inline shared/constants/security-provider.ts:17]: Besides the jsdoc why not create actual type and benefit from the full extent of TS? You could have something like:
```js
type SecurityProviderConfig = Record<
  SecurityProvider,
  {
    tKeyName: string;
    url: string;
  }
>;
```

**Review by cryptotavares — DISMISSED** [review]
Left a really small comment, but looks good!

File: shared/constants/security-provider.ts
Comment (digiwand) [reply shared/constants/security-provider.ts]: lgtm! I was following patterns from `shared/constants/gas.ts` and other files. updated here https://github.com/MetaMask/metamask-extension/pull/19825/commits/fac74730f8d9d889be872cde88bb9cfcfdcbb7b2

does this look okay or are there any more suggestions? 

cc: @cryptotavares

Comment (cryptotavares) [reply shared/constants/security-provider.ts]: [nit]
Do we want this const to be readonly (at least for TS)? If yes then we can:
```js
const SECURITY_PROVIDER_CONFIG: Readonly<SecurityProviderConfig>
```
or ditch the type and use const assertion ([TS docs for readonly and const assertion](https://www.typescriptlang.org/docs/handbook/typescript-in-5-minutes-func.html#readonly-and-const))
```js
export const SECURITY_PROVIDER_CONFIG = {
  [SecurityProvider.Blockaid]: {
    tKeyName: 'blockaid',
    url: 'https://blockaid.io/',
  },
} as const;
```

Using the const assertion seems more inline with what is being done on other constants TS files ([example](https://github.com/MetaMask/metamask-extension/blob/develop/shared/constants/app.ts#L110)).

(Sorry for asking to add the type and then replacing it for const assertion.. I am mostly used to use the Readonly mapped type, but that does not seem to be the standard on the Extension code 😅 )

**Review by cryptotavares — DISMISSED** [review]

File: ui/components/app/security-provider-banner-alert/security-provider-banner-alert.js
Comment (digiwand) [reply ui/components/app/security-provider-banner-alert/security-provider-banner-alert.js]: Hey @georgewrmarshall! hope you had a nice weekend. In addition to the lack of padding/margin between the title and the description, I realized there is another issue we have using the description BannerAlert/BannerBase property instead of passing children elements. 

The description currently expects a string property. The problem is that SecurityProviderBannerAlert will allow hyperlinks, or other HTML elements, in the description. 

error: 
> BannerBase   > checkPropTypes.js:20 Warning: Failed prop type: Invalid prop `description` of type `object` supplied to `BannerAlert`

  Do you have suggestions to address the two issues above?

Comment (digiwand) [reply ui/components/app/security-provider-banner-alert/security-provider-banner-alert.js]: In the meantime I have reverted the use of description param to fix the console and UI issues. I think it would be better to address this in a separate PR to unblock dependent work ahead 

Revert commit https://github.com/MetaMask/metamask-extension/pull/19825/commits/1c3d5508f00ad8f6f8e23ee42cd19e1c9aab2fda

Comment (digiwand) [reply ui/components/app/security-provider-banner-alert/security-provider-banner-alert.js]: I am thinking, maybe, we can do something similar to our plans with the Disclosure component and add support using Text as=“div”

File: shared/constants/security-provider.ts
Comment (digiwand) [reply shared/constants/security-provider.ts]: hey @cryptotavares, thanks for the suggestion and references! I applied the const assertion and I am thinking the type you suggested earlier is also nice to help define the types for each of the config objects. (Before, I was using jsdoc @typedef instead of the type you first proposed.)

Comment (cryptotavares) [reply shared/constants/security-provider.ts]: If you use the const assertion TS should automatically pick the correct type.
If you do const assertion and type the const at the same time, then TS will interpret the explicit type as the actual type (meaning that if you do not add the readonly utility type, then it will allow overriding the current property values). Example:
```js
// Just using const assertion
const test = {
  yolo: 1
} as const;

test.yolo = 2; // TS will complain about this because when we asserted to const, test is readonly
```

```js
type Test = {
  yolo: number
}

// explicit typing with const assertion
const test: Test = {
  yolo: 1
} as const;

test.yolo = 2; // TS will accept this, as it considers that test.yolo is a number and is not readonly
```
Anyway, I was suggesting that you can either use const assertion or you can explicitly type it (and use the Readonly utility type). But as const assertion seems to be the pattern used in the rest of the codebase you can do just that.

Btw, other examples on how we can force it to be readonly with explicit typing:
```js
// Using Readonly utility type ensures that all properties are readonly and cannot be reassigned. 
const test: Readonly<Test> = {
  yolo: 1
};
```
```js
const test: Test = {
  // using readonly modifier ensures that that specific property is readonly and cannot be reassigned
  readonly yolo: 1
};
```

**Review by georgewrmarshall — DISMISSED** [review]
LGTM! 🚀 
- Checked UI in storybook build

File: shared/constants/security-provider.ts
Comment (digiwand) [reply shared/constants/security-provider.ts]: ooh I see now. I think I got it https://github.com/MetaMask/metamask-extension/pull/19825 cc: @cryptotavares

**Review by cryptotavares — DISMISSED** [review]

File: ui/components/app/security-provider-banner-alert/security-provider-banner-alert.js
Comment (jpuri) [inline ui/components/app/security-provider-banner-alert/security-provider-banner-alert.js:98]: ??

**Review by jpuri — DISMISSED** [review]
🚀

File: ui/components/app/security-provider-banner-alert/security-provider-banner-alert.js
Comment (digiwand) [reply ui/components/app/security-provider-banner-alert/security-provider-banner-alert.js:98]: I thought it would be nice to separate the optional / not `isRequired` parameters

**Comment by digiwand** [comment]
@jpuri @cryptotavares @georgewrmarshall 

sorry had to force push to resolve a rebase conflict. Should be the same. I double-checked that the code works as expected in storybook. may I get a re-approval please?

**Review by jpuri — APPROVED** [review]

**Review by cryptotavares — APPROVED** [review]

**Pull request merged**
